tests/integration/cmd_link_unlink_test.go

```diff
@@ -104,9 +105,9 @@ var _ = Describe("odo link and unlink command tests", func() {
 			oc.WaitForDCRollout(dcName, project, 20*time.Second)
 			helper.HttpWaitFor(frontendURL, "Hello world from node.js!", 20, 1)
 
-			outputErr := helper.CmdShouldFail("odo", "link", "backend", "--component", "frontend", "--project", project, "--port", "8778", "--context", backendContext)
```

Comment:
we were using `backendContext` while we were trying to link to `backend`

Reply:
Have you verified the working of this one? It's using both `--component` and `--context` which means it's a hybrid of:
```
# Link component 'nodejs' to the 'my-postgresql' service
$ odo link my-postgresql --component nodejs
```
and
```  
# Link current component to the 'backend' component (backend must have a single exposed port)
$ odo link backend
```
which are kind of opposite ways of linking things.

Reply:
I'm asking because if someone were to quiz me which component will be attached to which in the line you've removed, I would not be sure and would have to rely on a guess.

Reply:
> we were using `backendContext` while we were trying to link to `backend`

The original command was correct but confusing. This is exactly the reason why we don't want to allow using `--context` and `--component`  flag together.

`odo link backend --component frontend --project  project --port 8778 --context backendContext`

Which means: take component defined in `backendContext` dir, and overwide the component name with `frontend` and project with `project` and link this to `backend`.
So this will actually link frontend to the backend.